system/metrics/sinks/elastic/elasticSink.hpp

```diff
@@ -0,0 +1,59 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#pragma once
+
+#include "jmetrics.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+
+//including cpp-httplib single header file REST client
+//  doesn't work with format-nonliteral as an error
+//
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#endif
+
+#undef INVALID_SOCKET
+#include "httplib.h"
```

Comment:
Why are these #includes needed in the header file?

The implementation should be private, and if the class implementation requires members, publish a factory method instead.

Reply:
I moved #include "nlohmann/json.hpp" to the cpp file since it may not be used for a class member variable.

However, I left httplib  since I currently plan on adding a protected member variable for a class defined in that header.

Reply:
As I move forward on the next step, it looks like a protected member variable will not be the direction. The include was moved to the spp file.